### PR TITLE
Set the kernel's default rp_filter setting for new interfaces to strict.

### LINKED ISSUE #806: Felix should check or set the default rp_filter setting for new interfaces
Felix now explicitly sets rp_filter to be enabled on the interfaces it controls but it'd be good to ensure that interfaces start with rp_filter enabled when they are created to close the window before felix sets the flag.

Simplest thing to do would be to set the default.rp_filter flag at felix startup.  We could also drop a file into /etc/sysctl.d/ in our deb/rpm install scripts (but best to do it in felix too since the install scripts aren't used by devstack, for instance).

### PR DESCRIPTION
**Pull request by fasaxc**
Closes a race with felix configuring the interface.

Fixes #806.

### CODE REVIEW

**Comment by alexwlchan** [comment]
I think the problem is line 51:

```python

**Comment by fasaxc** [comment]
Yep

**Comment by fasaxc** [comment]
Think I only ran the UTs I was expecting to break...

**Comment by fasaxc** [comment]
@matthewdupre Please can you review this?

File: calico/felix/devices.py [hunk 1]
```diff
@@ -54,6 +56,11 @@ def check_kernel_config():
                       "0 or 1.")
         raise BadKernelConfig("net.ipv4.conf.all.rp_filter set to 'loose'")
 
+    # Make sure the default for new interfaces is set to strict checking so
```
Comment (matthewdupre) [inline calico/felix/devices.py]: I think the log message is now a bit confusing.  It needs to be 1; just tell them to set it to 1.

I'm also a bit unsure about allowing setting 2 through.  Let's chat.

Comment (matthewdupre) [inline calico/felix/devices.py]: Oops, one is the 'all', the other is 'default'.  I think all that's needed is to add a note that 2 in all overrides specific configuration to 1; which we set for all interfaces (specifically and as the default for new interfaces).

**Comment by matthewdupre** [comment]
Oops, misread it.  I've suggested a smallish comment change, but looks fine otherwise.

**Pull request merged**